src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -687,14 +668,18 @@ - (void)_setDSTOffsets:(NSArray<MTRTimeSynchronizationClusterDSTOffsetStruct *>
 
 - (BOOL)_subscriptionsAllowed
 {
+    os_unfair_lock_assert_owner(&self->_lock);
+
+    // We should not allow a subscription for device controllers over XPC.
     return ![_deviceController isKindOfClass:MTRDeviceControllerOverXPC.class];
 }
 
 - (void)setDelegate:(id<MTRDeviceDelegate>)delegate queue:(dispatch_queue_t)queue
 {
     MTR_LOG("%@ setDelegate %@", self, delegate);
 
-    // We should not set up a subscription for device controllers over XPC.
+    std::lock_guard lock(_lock);
```

Comment:
Ideally we would not hold the lock across the call into testDelegate below...  It might be OK, though, since it's debug-only code.  Please add a comment down in that code, though, that if it ever stops being debug-only it needs to happen outside the lock.

Reply:
done.